### ISSUE TITLE
Electrolytic Separator vs Hydrogen Generator loop ( balance desired?) O2 only options?

### ISSUE THREAD

**Issue by Mementh** [aidancbrady/Mekanism#3041]
RE Issue: 
https://github.com/aidancbrady/Mekanism/issues/2743 2743
https://github.com/aidancbrady/Mekanism/issues/1037 1037

I would love it if this was re-examened and new options put in ... I feel there is a major imbalance in energy usage vs generation.

Right now the seperator does not benifit from the lower power usage that a energy upgrade gives I have a few ideas on that but would start with 1037

The new options to make power via bio fuel and hydrogen in 1037 i feel would be a good alternate path for power thats sustainable.
1.  I would say make the gas generator like a car engine.. very low power output 
2. using hydrogen as a battery i don't feel is a good idea.. there would be few uses of this in end game honestly or even in early game.

3. i like the idea of a mid game using the ethelyn and hydrogen.. I feel this should be a ok but medium sized on the setup.. a few blocks.. a multi block structure ?

4. the fusion reactor currently sits as the king imho of power and we don't need more then that long term for extreme power.. 

Why do i want this.. a seperator using 20k RF/tick is silly..  but i know things have had a reason.. logic behind it to prevent the hydrogen loop issues other things had.

a few ideas if it could be done... 

a late game catalyst of some kind that you make a "factory" that allows oxygen to be extracted better from water more effectivly?

Could use platnium/shiney metal from thermal foundry or nickle iron mix? ( add to ore generatrion there metal so its easily compatable?) This would be made out of a PRC and a seperator and other high end alloys

There could be two settings for the machine.. one for just oxygen making.. and one for hydrogen 

Set it to Oxygen and it just works with water.. no other things needed. no hydrogen output!
Set it to Hydrogen it will need biomass as the PRC does but no oxygen output!
( if just oxygen https://en.wikipedia.org/wiki/Electrolysis_of_water lists there are more then a few ways to make it more efficent AKA heating the water usign the hydrogen as heat so its just a O2 Maker )

Another option is a machine that could pull out of the air AKA galacticraft does? This would "cool" the air and make it liquid so you would need a rotary machine to make it a usable gas.  ALA https://en.wikipedia.org/wiki/Air_separation ( this would respond to energy upgrades)

another option .. nerf the generator so that it at max energy reduction of a seperator gives the same ammount of energy from the hydrogen as todays seperator?  if thats the exploit issue... thats the simplest option right? 
i will agree 100% it does not meet chemical real world options but would be the simpleist to add into game and would allow the server moderator to set the energy output to current "normal" if they did not mind the explit ( just like they can now)

TLDR: Better use of hydrogen is AWESOME. Make a new machine to make O2 only! or make hydrogen generators use more to make same energy to balance

**Comment by VT-14** [comment]
My 2 dollars (I passed 2 cents 10 minutes ago):

If I recall High School Chemistry correctly, the combustion of hydrogen is:
2H_2 + O_2 -> 2H_2O  
(2 hydrogen gas molecules + 1 Oxygen gas molecule makes 2 water molecules)

What the Mekanism Electrolytic Separator (and a real one) does is the opposite of that reaction.  [Now pulling on College Physics]  To reverse the reaction, you have to provide (at an absolute minimum), the energy you would have gained in the reaction.  In the real world, you have energy losses, so you have to pour in even more energy.

If you could get more energy, you would break the laws of Thermodynamics and Conservation of Mass/Energy.  You would also make a fortune by making a perpetual motion machine and shattering Physics.

Simply put: you can't make energy by simply pulling water apart and putting it back together.

That is why Ethylene is in Mekanism.  I am not qualified to analyze the reactions involved in making and using Ethylene, but I would be quite confident in saying that the energy produced is from the Bio-Fuel, and the process of making the Ethylene just improved the efficiency.


That said, Mekanism doesn't exactly parallel real life (nor does it have to).  The current, non-upgraded Electrolytic Separator is 100% efficient; you can burn the Hydrogen for as much power as it took to make it.  What I see as the problem is that the efficiency goes down as you give it speed upgrades, and it's impossible to counter that with Energy Upgrades.  The Electrolytic Separator is the only Mekanism machine that I know does this (due to the exploit), and I would suggest that the functionality be changed to where an energy upgrade will only lower energy cost while the same or a greater number of speed upgrades are installed to preserve a possible 100% efficiency with an equal number. 

I would prefer such a balance only applying to the Electrolytic Separator (a furnace running at 2 RF/t is a nice option), but it would be fair for this to be applied universally.

Examples:
Speed (multiplier) | Energy (multiplier) | Power (RF/t) | Efficiency
0 (1x) | 0 (1x) | 160 RF/t | 100% (1 power for 1 fuel)
8 (10x) | 0 (1x) | 16000 RF/t | 10% (100 power for 10 fuel)
0 (1x) | 8 (negated) | 160 RF/t | 100% (1 power for 1 fuel)
8 (10x) | 8 (10x) | 1600 RF/t | 100% (10 power for 10 fuel)

**Comment by Mementh** [comment]
VT-14 awesome!! it feels like the balance would be kept! and would be simpler then adding in NEW machines to fix the imbalance?

I do like the idea of the PRC to make Ethylene to make more power. 
I did setup a deal to convert SOME of the excess H2 into Ethylene  but getting the biomass is harder then thought. since 2 9x9 wheat farms don't keep things stocked up :/ 

I would suggest possably allowing all food from pamscraft to be added in as sources for bio fuel.. also combining the bio reactor and gas generator to make just a generator that if given gas give X energy and biomass Y energy and Ethylene Z energy 

Another idea is energy and speed upgrades inthe generators.. increased buffer they can hold.. !not effecciancy at all! speed just makes it burn through faster giving more energy till buffer is full.

**Comment by rdunham3** [comment]
An option in the configurations to add other mods' bio-based items as crusher biofuel recipes would do the trick and be least effort for Aidan & co.  I find myself wanting to throw all sorts of things in there because they're biological and otherwise useless.  Mod saplings, Wuppy's peaceful pack flax, vanilla spider eyes...

I used to think the gas generators were a bit under-powered but in my latest world I started by going for a crusher & PRC for ethylene production ASAP and a 9x9 mini-farm to supply it.  I found that it kept up with my energy needs even with a 3x ore processing setup, provided I didn't use any speed upgrades, which I generally don't in my first base.  The balance is great!  You only run into problems when you start overclocking the system.  For me, early on, the answer is to be patient and go mine while the mini-farm is growing.  When I start building the 2nd base, the one with an evaporation tower, then I can upgrade machines to use up some of that extra power the tower provides. :)

**Comment by unpairedbracket** [comment]
@VT-14 The current (intended at least) behaviour of the separator is that it has constant efficiency (200J/1mB water electrolysed with the configs in my test env). Energy upgrades increase storage up to 10x, speed upgrades increase processing speed and energy usage by the same multipliers (therefore keeping efficiency constant), up to I think 256x speed. Your idea is interesting though, I'd definitely be interested in implementing something like that. I would worry that people may complain about energy upgrades having no effect without speed upgrades, but then again people are already complaining, so what do I know?

@Mementh I like the idea of a chemical addition to the electrolysis (not strictly a catalyst, as they're not used up in the reaction, so can't give an energy input) to allow it to happen more efficiently than what would be 100% with just the voltage across the water. I would suggest one of two things:
* My first suggestion would be to allow an optional item input (some kind of chemical) to the separator that either reduces energy input or improves yield, thus making hydrogen loops energy-positive again, as the energy is extracted from the chemical as it's used up.
* Then I realised we can just react water with lithium and get lithium hydroxide and and hydrogen out. Lithium already has a process, and we can just ignore the hydroxide product (although it is used in CO2 scrubbers in stuff like rebreathers...) and make this a Chemical Infuser recipe. I've been too busy to work on Mekanism very much for the past few months, so I can't remember how much of a pain it is producing lithium, but might this be a viable way of producing hydrogen?

For production of oxygen there's the Ambient Accumulator, a machine I've been working on, but which I don't think is enabled in release builds yet. It collects an ambient gas from the air, and will collect oxygen in the Overworld. Requires no power, and collects fun interesting gases in other dimensions. Hopefully that should help with O2 issues.

**Comment by Mementh** [comment]
@unpairedbracket hugs! i appreciate the look.. I do think @VT-14 thought that it should always be 1:1 but right now feels off at the higher end... also that the generators are fixed on output so you need Much much more of them phsyically.. :/ 



Ambient Accumulator sounds awesome and i don't mind if it uses some power.. unless its meant to be like the aquaious accumulator that just makes water from surrounding without power?  Also what would the other gases be that it collects i wonder? 




setting up to get lithium is hard! it requires most of whats needed for the reactor setup other then the solar neutron activators... 

It requires a solar evaporation to make brine then another solar evaporataion to process the brine into lithium then a condensorator to make it a gas then a crystaliser to make it a solid (why can only gasses go into the crystaliser? ) 

I would suggest somehow making this a "closed loop" system where the lithium is just introduced and when burned returned? so it a carrier of the energy/only a catalyst? 

the problem i feel here is lithium is near end game in my mind.. closer to the reactor side which invalidates the need for this as a power generator for the most part. 




So a replacement is needed that would be something that oxidises in a chemical oxidizer and mixes with hydrogen in a infuser and that gas burns for a decent clip in a new generator?

I would think this new generator should be closer to early game but near middle.. and replace gas generators and biofuel as one type of generator ( remove both of them from the modpack)?? ( allow speed upgrades to burn "faster" but not increase fuel usage so that you don't need as many and reduce server stress?)
It should be a multi block structure maybe? 2x3 at most?  able to put out 3-5k/tick at full production? with speed maybe 20k-30k/tick  
I don't feel sticking with the 8 speed = 256x would be wise here at all! 


I also would assume anyone at this stage has enough to start working with ender chests or tesseracts for power remotely ( or if you guys have a repalcement for that ) so that they dont need to haul this around.


I had a longer part of the fusion reactor but realized you guys had done a good job of figuring that out brine to lithium to tritium..  which makes sense now with deutrium. *woot*

**Comment by VT-14** [comment]
Well, I had to run a test.  It turns out that my information was out of date.  The setup in Mekanism 1.7.10-8.1.7.252 (may not be current, but is certainly closer) has it so that the Energetic Infuser is always 100% efficient.  A filled Basic Energy Cube's worth of power will always make 10,000 Hydrogen, which can be burned to exactly fill a Basic Energy Cube.

However, it still feels off.  The standard effects of speed and energy have it so that increasing speed decreases efficiency, and adding power increases efficiency, and they will balance at equal numbers.  I realize that Energy Upgrades also increase the power capacity of a machine, but I have personally never sought an upgrade for that alone.  The Energetic Infuser being the only exception to both of the efficiency rules is weird.  If this method is the permanent solution, then I would suggest having it altered by Gas Upgrades or a new upgrade instead.


A Generator Upgrade (or Factory equivalent) would be greatly appreciated as well.  I see it as most needed for Gas-Burning Generators where to utilize something like Ethylene nearly as fast as you can make it with an upgraded setup would require hundreds of generators (and I was getting a notable amount of lag when trying that in a test world).  Bio-Generator and the non-passive use of the Heat Generator would be fair game too.  In my opinion, the purely passive generation of Wind Turbines and Solar Generators (and Heat-Generators) shouldn't make use of this upgrade.  The Reactor can just increase the reaction rate.

**Comment by unpairedbracket** [comment]
@Mementh I thought that might be an issue with using lithium here. Another proposition: iron is also more reactive than hydrogen, but not to the extent that lithium is. Lithium is reactive enough that it spontaneously reacts (quickly) with water and iron isn't, but it's conceivable (given I'm a physicist, not a chemist; please no chemists shout at me) that adding iron dust to the separator could allow some sort of galvanically catalysed oxidation of the iron, giving us hydrogen gas out as a product and needing less electrical energy input as a result. I think
2 H<sub>2</sub>O + Iron Dust + Energy → 2 H<sub>2</sub> + O<sub>2</sub> + Rust? → 2 H<sub>2</sub>O + Rust? + Lots More Energy
 seems a bit more realistic than just
2 H<sub>2</sub>0 + Energy → 2 H<sub>2</sub> + O<sub>2</sub> → 2 H<sub>2</sub>0 + Lots More Energy
Question mark on rust is because rusting of iron _is_ exothermic, but doesn't actually produce H<sub>2</sub> as a product. The rusting equation is 
4 Fe + 3 O<sub>2</sub> + 2 H<sub>2</sub>O → 4 FeO(OH) → 2 Fe<sub>2</sub>O<sub>3</sub> + 2 H<sub>2</sub>O
but I'd have no qualms about just inventing an electrochemical reaction that does something like
2 Fe + 3 H<sub>2</sub>O → Fe<sub>2</sub>O<sub>3</sub> + H<sub>2</sub>
when iron dust in water has a potential difference put across it. (The opposite is a real thing, and is endothermic, so this reaction would give us a net output of energy). Iron is fairly easy to get hold of, and dust is one of the easiest forms to obtain. Would this be a good compromise?


@VT-14 I agree that it still feels a bit not quite right, but I'm not quite sure what to do to improve it. I'll keep on thinking about the issue and see if I can come up with anything more satisfactory, without having the effect of energy upgrades negated depending on number of speed upgrades, which I suspect would also annoy/confuse people.

I feel like a generation speed upgrade (that increased consumption rate of whatever fuel is used as well, obviously) would possibly be a good idea, but I suspect we would want to be careful to avoid issues. No idea what those issues would be, but there are usually issues.

**Comment by Nova001Sol** [comment]
@unpairedbracket Tell me if I'm wrong here, but your reaction would be akin to the steam-iron process, yes? If so, it definitely can be done if the water is heated first. I suggest you take a look at this: https://en.wikipedia.org/wiki/Lane_hydrogen_producer

**Comment by aidancbrady** [comment]
I've made some changes in v9 that allow for Gas Generators to scale gas processing to match the input, to a maximum of 100 units of gas to be consumed per tick.

Furthermore, after doing some testing, I can confirm that the water -> electrolytic separator -> hydrogen process is 100% efficient, no matter how many speed upgrades you have installed.  As such, I'm going to market this closed.
